File: HA3_4.ts
//Write a program which accept number from user and display all its non factors
//Input : 12
//Output: 5    7    8   9   10  11

import * as readline from 'readline';

function SumNonFact(iNo : number)
{
    let iCnt : number = 0
    let Ans : number = 0

    for(iCnt =1; iCnt < iNo; iCnt++)
    {
        if((iNo % iCnt)!=0)
        {
            Ans = Ans + iCnt;
        }
    }
    return Ans;
}

async function main()
{
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    })

    let iValue : number = 0
    let iRet : number = 0;

    iValue = parseInt(await askQuestion(rl, "Enter a number")) ||0

    iRet = SumNonFact(iValue);

    console.log("SUmmation is : "+ iRet);

    rl.close();
}

function askQuestion(rl: readline.Interface, question: string): Promise<string>
{
    return new Promise((resolve) =>{
        rl.question(question,(answer : any) =>{
            resolve(answer)
        })
    })
}
main()

File: HA3_4.cpp
//Write a program which accept number from user and return summation of all its non factors
//Input: 12
//Output: 50

#include<iostream>
using namespace std;

int SumNonFactors(int iNo)
{
    int Add =0;
    int iCnt = 0;

    for(iCnt = 1; iCnt<iNo; iCnt++)
    {
        if((iNo % iCnt)!=0)
        {
            Add = Add+ iCnt;
        }
    }
    return Add;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter a number"<<"\n";
    cin>>iValue;

    iRet = SumNonFactors(iValue);

    cout<<"Summation is : "<<iRet<<"\n";

    return 0;
}